### PR TITLE
Added some basic tests

### PR DESCRIPTION
**Pull request by wibblymat**
R: @jeffposnick @jakearchibald

I never found a satisfactory way of running the tests in the worker, but hopefully this will work.

Doesn't currently cover all features, but some is better than none.

### CODE REVIEW

File: tests/tests.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  QUnit.test('First declared route wins', function(assert) {
+    // Matches both routes
+    checkValue('multiple/match/anything.html', '1', assert);
+
+    // Only matches second route
+    checkValue('multiple/match/anything', '2', assert);
+  });
+
+  QUnit.test('Caching', function(assert) {
+    var date = +(new Date());
```
Comment (jeffposnick) [inline tests/tests.js]: `Date.now()` would be clearer.

File: tests/tests.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    var url = 'cache/' + date;
+    var done = assert.async();
+    // Empty cache
+    fetch(url).then(function(response) {
```
Comment (jeffposnick) [inline tests/tests.js]: I'm having a hard time following the flow here, and understanding what each step is performing.

Could you a) break out the rejection handlers into a separate `.catch()` if possible (that tends to make things more readable), and b) add some comments?

File: tests/index.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html>
+  <head>
+    <title>Service Worker Toolbox Tests</title>
+    <link rel="stylesheet" href="//code.jquery.com/qunit/qunit-1.17.0.css"/>
```
Comment (jeffposnick) [inline tests/index.html]: Can you pull in `qunit` locally via a `devDevependency` in your `package.json` and use that? I'm sure `code.jquery.com` isn't going away anytime soon, but I don't see the benefit of depending on something remote.

**Comment by jeffposnick** [comment]
Thanks for writing tests! They look good in general.

My main piece of feedback is that the tests would be more useful if they could be automated via some sort of runner, which was then tied in to the release process. I've had good luck using `selenium` to test the `<platinum-sw>` elements in supported browsers via the [`web-component-tester`](https://github.com/Polymer/web-component-tester) framework, and maybe that's something you could borrow (without all the web component-y bits).

I don't think that needs to block merging this PR, but automation could be a future enhancement.

**Comment by wibblymat** [comment]
Definitely agree about automation! I'd love to run this through Travis. I've opened a separate issue for that (#15).

Will resolve the other issues.

**Comment by addyosmani** [comment]
> My main piece of feedback is that the tests would be more useful if they could be automated via some sort of runner, which was then tied in to the release process.

@jeffposnick are you familiar with our Polymer CI runner? http://polymerlabs.github.io/ci-runner/#!/PolymerElements/iron-a11y-keys/1ab9e69aff38a8712dffdfe6749a6d83fcab9e84

One of the benefits is that it's able to run your WCT tests through CI each time a PR comes in, which can be handy for making sure your tests aren't breaking with new patches coming through.

**Comment by jeffposnick** [comment]
@addyosmani Yep, I am! It would be awesome if Mat could move towards something like that, though WCT as-is might not be an option, since his tests are QUnit-based and he doesn't have a web component.

I'm not entirely happy with the [`sw-precache` approach](https://github.com/GoogleChrome/sw-precache/blob/master/test/test.js), but it's another option—I use `mocha` to test the behavior of various JS helper methods, but don't test the actual service worker in-browser behavior.

**Comment by wibblymat** [comment]
@jeffposnick PTAL

**Comment by jeffposnick** [comment]
Not sure if you're opposed to pulling in QUnit as a local dependency or if you just missed that comment. If you'd rather continue using the CDN version, that's fine too, and feel free to merge.

**Comment by addyosmani** [comment]
My two cents are a local dep for QUnit would make sense here, but if the CDN version is preferable and we intend to move over to something else long term anyway, that also sgtm. 

PR itself LGTM.

**Comment by wibblymat** [comment]
Sorry, actually missed the qunit comment first time around. Have brought it in locally. Will merge, but feel free to check over the latest commit :)

**Comment by addyosmani** [comment]
:+1:

**Pull request merged**